### DISCUSSION TITLE
Style and Composition transfer

### DISCUSSION THREAD

**Discussion by cubiq** [cubiq/ComfyUI_IPAdapter_plus#376] [Announcements]
I'll try to use the Discussions to post about IPAdapter updates.

I just pushed an update to transfer Style only and Composition only. You find the new option in the `weight_type` of the advanced node. The style option (that is more solid) is also accessible through the Simple IPAdapter node. It works only with SDXL due to its architecture.

For the composition try to use a reference that has something to do with what you are trying to generate (eg: from a tiger to a dog), but it seems to be working well with pretty much anything

|![composition_1](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/427614/23999bce-d204-4ef9-a6d5-0664ab243956)|![composition_3](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/427614/42ab2851-3eee-41af-b4eb-d484233d761b)|
|---|---|

The style transfer works very well and can be improved with accurate prompting. In the corner you can see the reference image.

|![style_transfer_3](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/427614/a4f3935a-fa17-4fe3-b3e9-075ae5e15add)|![style_transfer_1](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/427614/0baf28ab-704a-48ef-b3ac-cebc7f903083)|
|---|---|

When I have time I'll try to add a node that lets you pick both the style and the composition image.

**Comment by zhangn77** [comment]
It's a wonderful work! Can I ask how to only refer the composition from the target image by IP-Adapter?

**Reply by cubiq** [reply]
set the weight type to "composition"

**Reply by zhangn77** [reply]
Sorry, I want to ask that the algorithms behind this component, could you please explain it? Because as I know, the IP-Adapter cannot decompose the composition from the target image.

**Reply by cubiq** [reply]
In SDXL the unet layers are very specialized, so I basically apply the weights only to specific layers

**Reply by zhangn77** [reply]
thank you! wonderful idea! Can it apply to sd 1.5 unet?

**Comment by xiaohu2015** [comment]
Well done, you made IP-Adapter great again!

**Reply by cubiq** [reply]
hehehe IPAdapter has always been great :smile: 

Have you tried to play with the blocks and the Face models? I don't seem to be able to find the right ones.

Here more examples of style and composition

![GKQtqoUXIAANYfh](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/427614/6e590939-63e7-4f8a-837f-dbe3e2486aaf)

![GKQtqoZXYAAKisz](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/427614/dd09adae-d4c8-4598-9604-da5913c9089a)

![GKQZ1tAXsAECN5e](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/427614/048aa559-6d6c-4b99-af2a-24659126e0d1)

**Reply by xiaohu2015** [reply]
face is only part of the image, maybe not work, maybe need to combine different blocks

**Comment by cubiq** [comment]
It seems to be possible for SD1.5 too. I need more time for testing (images upscaled by 1.5)

| ![style_transfer_9](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/427614/0c88dff1-1c1c-483b-b84d-6729d9ca7eef) | ![style_transfer_8](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/427614/b529cced-9680-4990-8583-4d45f5f9f53f) | ![style_transfer_7](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/427614/b1006034-21e8-486c-b644-eba3ca832f44) |
|---|---|---|

**Comment by benj-pan** [comment]
![image](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/166602243/ac79469b-569b-49b7-a452-27260817eebe)
This feature excites me. However, I installed the latest version as you provided, but I don't find SDXL in weight_type.

**Reply by benj-pan** [reply]
May I ask how to solve this?
